cryptoki/src/types/slot_token.rs

```diff
@@ -42,3 +42,45 @@ impl From<Slot> for CK_SLOT_ID {
         slot.slot_id
     }
 }
+
+///
+#[derive(Debug, Clone, Copy, Default)]
+#[repr(C)]
```

Comment:
I'm not sure we need this here. I'm thinking we could make `TokenInfo` a proper abstracted version of the FFI type, and simply have conversions between them. This would allow us to, for example, present `label`, `manufacturer_id` and `model` as `String` directly.

Reply:
Agree with that, when we use `repr(C)` or `repr(transparent)` in `cryptoki` is when the Rust abstracted structures must match the memory layout of their C variants. The main reason for that is that then we don't need to do any move of the data owned by the application. I can expand on that if you want, took us a while to find out through a lot of different trials and errors...
In the case of `CK_TOKEN_INFO`, it is only used for `C_GetTokenInfo` and does not need to contain any initial data from the application. That means that we can call `C_GetTokenInfo` with an empty `CK_TOKEN_INFO` and then convert that to an abstracted, Rust native, `TokenInfo` which does not need to have any specific memory layout.

There are multiple ways of how to design `TokenInfo`:
- for each field have an abstracted Rust version (`CK_UTF8CHAR`/`CK_CHAR` -> `String`, `CK_FLAGS` -> `Flags`, `CK_ULONG` -> `Ulong` but could also probably be `u32`, etc...). All the fields are put `pub`. That way we have to convert everything when `TokenInfo` is created.
- have `TokenInfo` contain a `CK_TOKEN_INFO` as a private field and have getters that perform the conversion dynamically.

I think the second option is better as it easy to add and we can only add the getters when we need them.

Reply:
Okay, done that for manufacturer_id and serial_number (that I need). If someone needs more they can add appropriate functions.